python/merkle_proofs/smtr.py

```diff
@@ -17,4 +18,23 @@ def sign_tree_root(root: bytes, tree_alg: ta.TreeAlgorithm, key, alg, tree_size=
     
     msg = Sign1Message(phdr=phdr, payload=root)
     msg.key = key
-    return msg.encode(tag=True)
+    signed = msg.encode(tag=True)
+    if not detached:
+        return signed
+    detached = detach_cose_sign1_payload(signed)
+    return detached
+
+
+def detach_cose_sign1_payload(msg):
+    # pycose doesn't support detached payloads, so we have to do it manually.
```

Comment:
sadge

Reply:
Won't be necessary with the next release: https://github.com/TimothyClaeys/pycose/pull/110